ppafm/cli/gui/ppafm_gui.py

```diff
@@ -218,6 +218,7 @@ def scanWindowFromGeom(self):
         '''Infer and set scan window from current geometry'''
         if self.xyzs is None: return
         scan_size = self.xyzs[:, :2].max(axis=0) - self.xyzs[:, :2].min(axis=0) + 2 * self.sw_pad
+        scan_size = np.minimum(scan_size, [25, 25]) # Let's not make it automatically too large, so that we don't run out of memory
```

Comment:
This only sets the initial scan size when you load a file. You can set whatever size in the GUI after the file is loaded. 

Initially it takes the bounding box of the geometry and adds some padding to get the scan size, but if you have a very large system (esp. a slanted surface slab), then you could run out of memory and crash before you even get to the GUI, and you never even have the chance to set a reasonable scan size.

Reply:
This is just some quick fix for a problem I was having. Really we should have some system of tracking the amount of memory that is in use: #79.

Reply:
> Really we should have some system of tracking the amount of memory that is in use: #79.

that would be great. Do you know about some possibilities how to do that? E.g. ask the GPU driver of OpenCL how much memory you allocated? I was searching for it long time agou, but did not find much usefull.

Reply:
I only know that you can query the maximum memory, but I don't know about the current memory usage. It's probably something that we need to do manually anyways, since the program automatically cannot know how much memory an operation takes before you do it. I'm thinking for example about an FFT, where we would need to estimate how much memory is required during the operation before we do it so that we don't crash in the middle of it.

Reply:
> It's probably something that we need to do manually anyways, since the program automatically cannot know how much memory an operation takes before you do it. 

But we the programmer also cannot know how much memory other processes already allocated before you. So it would be really usefull to be able to do something like system task-manager are doing. I mean e.g. like `top` in linux.

```
top - 16:47:53 up  7:43,  1 user,  load average: 0.37, 0.32, 0.30
Tasks: 369 total,   1 running, 368 sleeping,   0 stopped,   0 zombie
%Cpu(s):  1.9 us,  0.8 sy,  0.0 ni, 97.3 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
MiB Mem :  15844.8 total,   4660.6 free,   6849.3 used,   4334.9 buff/cache
MiB Swap:      0.0 total,      0.0 free,      0.0 used.   8356.9 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                          
   1043 root      20   0   24.4g 201964  41256 S   7.0   1.2   6:08.10 Xorg                                             
   2245 prokop    20   0   32.7g 521892 261632 S   6.3   3.2  13:37.91 chrome                                           
   2289 prokop    20   0   32.7g 849836 218364 S   4.6   5.2  21:38.22 chrome                                           
   1539 prokop    20   0  502056  83112  51980 S   3.3   0.5   3:19.92 xfwm4                                            
   1561 prokop    20   0  470136  38928  29216 S   2.3   0.2   0:17.87 xfce4-panel                                      
  30223 prokop    20   0 1130.0g 164800 107636 S   1.7   1.0   0:10.08 chrome                                           
  30624 prokop    20   0  491184  51444  39420 S   1.7   0.3   0:00.48 xfce4-terminal                                   
   1570 prokop    20   0  596464  93496  31348 S   1.3   0.6   0:09.28 xfdesktop                                        
   2291 prokop    20   0   32.4g 132340  93916 S   1.3   0.8   6:44.86 chrome                                           
    651 root     -51   0       0      0      0 S   1.0   0.0   2:37.83 irq/158-nvidia                                   
   1736 prokop    20   0 7380320 327764  88788 S   1.0   2.0   3:22.91 dropbox                                          
   1577 prokop    20   0  443844  52140  35736 S   0.7   0.3   0:06.76 panel-30-xkb                                     
   1588 prokop    20   0  672228  46268  35956 S   0.7   0.3   0:44.19 panel-21-pulsea                                  
    487 root     -51   0       0      0      0 S   0.3   0.0   0:01.65 irq/149-iwlwifi              
```